packages/rpc/src/tests/utilities.ts

```diff
@@ -1,13 +1,35 @@
 class MessagePortPolyfill implements MessagePort {
-  onmessage: EventListener | null = null;
   onmessageerror: EventListener | null = null;
 
   otherPort!: MessagePortPolyfill;
   private listeners = new Set<EventListener>();
 
+  // MessagePort does not send messages unless it is started via start() or attaching .onmessage
+  // https://developer.mozilla.org/en-US/docs/Web/API/MessagePort/start
+  private started = false;
+  private _onmessage: EventListener | null = null;
+
+  // If the port is not yet started, messages will be queued for sending.
+  private messageQueue: Event[] = [];
```

Comment:
Might make more sense to call this `eventQueue`, since there could technically be events other than message events? Or, if it really can only cover message events, maybe we should be doing the guarding in `postMessage` instead of `dispatchEvent`?